docs/Developers/Debug_Bears.rst

```diff
@@ -122,3 +122,88 @@ Or to debug all bears specified by ``bears`` setting:
     bears = PEP8Bear,MypyBear
     files = <filename>
     debug_bears = True
+
+Debugger features
+-----------------
+
+Inspect Bear settings
+^^^^^^^^^^^^^^^^^^^^^
+
+New command ``settings`` included in coala's Debugging Interface to inspect
+Bear settings in debugging environment, ``settings`` command displays the
+updated setting of a Bear i.e., user can quickly inspect the passed settings.
+
+Below is the code of simple TooManyLinesBear which detects if a file has more
+than ``max_number_of_lines`` lines.
+
+.. code:: python
+
+    from coalib.results.Result import Result
+    from coalib.bears.LocalBear import LocalBear
+
+
+    class TooManyLinesBear(LocalBear):
+
+        def run(self, filename, file, max_number_of_lines: int):
+            """
+            Detects if a file has more than ``max_number_of_lines`` lines.
+
+            :param max_number_of_lines:
+                Maximum number of lines to be allowed for a file.
+            """
+
+            if len(file) > max_number_of_lines:
+                yield Result.from_values(
+                    origin=self,
+                    message=('This file has {} lines, while {} lines'
+                             ' are allowed'
+                             .format((len(file)), max_number_of_lines)),
+                    file=filename)
+
+In the above example ``max_number_of_lines`` is non optional setting of a
+TooManyLinesBear. After invoking debugger on TooManyLinesBear coala will first
+ask the value of all unpassed non optional setting i.e, ``max_number_of_lines``
+and after that debugger will get invoke on ``run()`` method of TooManyLinesBear,
+``settings`` command will be used to display the updated optional and non
+optional settings of a Bear in debugging environment i.e.,
```

Comment:
I'm not sure if it's relevant to explain again the whole coala process.

Reply:
I wanted that user get familiar with coala process and can easily understand the settings part of 
`debugger`, If you will say I will remove the above 2 line :+1: . For now I am updating without removing it.